src/test/java/TestOrderDaoImpl/CustomerDaoSecondImplTest.java

```diff
@@ -0,0 +1,80 @@
+package TestOrderDaoImpl;
+
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+import workwithJDBC.CustomerDaoSecondImpl;
+import workwithJDBC.Customers;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import static org.junit.Assert.assertTrue;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class CustomerDaoSecondImplTest {
+
+    private static CustomerDaoSecondImpl customersDaoSecond;
+    private static Customers customer1;
+    private static Customers customer2;
+
+    @BeforeClass
+    public static void beforeClass() {
+        customersDaoSecond = new CustomerDaoSecondImpl();
+        customer1 = new Customers();
+        customer2 = new Customers();
+        customer1.setCustNum(BigDecimal.valueOf(3333));
+        customer1.setCustRep(BigDecimal.valueOf(103));
+        customer1.setCreditLimit(BigDecimal.valueOf(888));
+        customer1.setCompany("Wood");
+        customer2.setCustNum(BigDecimal.valueOf(4444));
+        customer2.setCustRep(BigDecimal.valueOf(104));
+        customer2.setCreditLimit(BigDecimal.valueOf(400));
+        customer2.setCompany("Gold");
+    }
+
+    @Test
+    public void Test1InsertCustomer() throws SQLException {
+        String sql = "INSERT INTO customers (cust_num, company, cust_rep, credit_limit) VALUES (?, ?, ?, ?)";
+        assertTrue(customersDaoSecond.insertCustomer(customer1, (customers, conn) -> {
+            PreparedStatement stmt = conn.prepareStatement(sql);
+            stmt.setBigDecimal(1, customer1.getCustNum());
+            stmt.setString(2, customer1.getCompany());
+            stmt.setBigDecimal(3, customer1.getCustRep());
+            stmt.setBigDecimal(4, customer1.getCreditLimit());
+            return stmt;
+        }));
```

Comment:
Why your realisation of methods in Test? It must be in CustomerDaoSecondImpl. In test you should only to call this methods.

Reply:
Yes. It critical. Your template must be as part of logic